File: lib/src/main/java/ir/arefdev/irdebitcardscanner/DebitCard.java
package ir.arefdev.irdebitcardscanner;

public class DebitCard {

	public final String number;
	public final int expiryMonth;
	public final int expiryYear;

	public DebitCard(String number, int expiryMonth, int expiryYear) {
		this.number = number;
		this.expiryMonth = expiryMonth;
		this.expiryYear = expiryYear;
	}

	public String last4() {
		return number.substring(number.length() - 4);
	}

	public String expiryForDisplay() {
		if (isExpiryValid()) {
			return null;
		}

		String month = String.valueOf(expiryMonth);
		if (month.length() == 1) {
			month = "0" + month;
		}
		String year = String.valueOf(expiryYear);
		if (year.length() == 4) {
			year = year.substring(2);
		}

		return month + "/" + year;
	}

	public boolean isExpiryValid() {
		return expiryMonth <= 0 || expiryMonth > 12 || expiryYear <= 0;
	}
}

File: lib/src/main/java/ir/arefdev/irdebitcardscanner/DebitCard.kt
package ir.arefdev.irdebitcardscanner

class DebitCard(@JvmField val number: String, @JvmField val expiryMonth: Int, @JvmField val expiryYear: Int) {

    fun last4(): String = number.substring(number.length - 4)

    fun expiryForDisplay(): String? {
        if (isExpiryValid()) return null

        var month = expiryMonth.toString()
        if (month.length == 1) month = "0$month"
        var year = expiryYear.toString()
        if (year.length == 4) year = year.substring(2)

        return "$month/$year"
    }

    fun isExpiryValid(): Boolean = expiryMonth !in 1..12 || expiryYear <= 0
}
